CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_SquadPickupPoint.uc

```diff
@@ -0,0 +1,218 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  statusNone
+//  PURPOSE: This class creates a dummy Geoscape entity allowing HQ to fly over
+//           in order to exfiltrate troops from an ongoing covert action
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class XComGameState_SquadPickupPoint extends XComGameState_GeoscapeEntity;
+
+var StateObjectReference ActionRef;
+
+var StrategyCost ExfiltrateCost;
+
+var bool bSkyrangerDeployed;
+var bool bSquadExfiltrated;
+// this is a flag to assure we only use it once
+var bool bConsumed;
+
+function bool RequiresAvenger()
+{
+	return !bSkyrangerDeployed;
+}
+
+function bool RequiresSquad()
+{
+	return false;
+}
+
+function SetupPickupLocation(XComGameState NewGameState, XComGameState_CovertAction CovertAction, StrategyCost Cost)
+{
+	local XComGameState_HeadquartersXCom XComHQ;
+	ActionRef = CovertAction.GetReference();
+	// we're intentionally checking if this Region has a XCGS and assigning none if not
+	Region = XComGameState_WorldRegion(`XCOMHISTORY.GetGameStateForObjectID(ActionRef.ObjectID)).GetReference();
+
+	ExfiltrateCost = Cost;
+
+	Location.x = CovertAction.Location.x;
+	Location.y = CovertAction.Location.y;
+	
+	XComHQ = `XCOMHQ;
+
+	XComHQ = XComGameState_HeadquartersXCom(NewGameState.ModifyStateObject(class'XComGameState_HeadquartersXCom', XComHQ.GetReference().ObjectID));
+	XComHQ.CrossContinentMission = GetReference();
+}
+
+function FlyToPickupPoint()
+{
+	// if we have a region fly there
+	if (Region.ObjectID != 0)
```

Comment:
this check will always fail (see [this](https://github.com/WOTCStrategyOverhaul/CovertInfiltration/pull/128/files/c460c193d15c4d9f9d23b3599dd3ba277cbd5fd7#r263023055) comment) which leads me to believe that the whole `bSkyrangerDeployed` workaround is not required